llama-index-integrations/vector_stores/llama-index-vector-stores-weaviate/llama_index/vector_stores/weaviate/base.py

```diff
@@ -143,11 +147,13 @@ class WeaviateVectorStore(BasePydanticVectorStore):
     auth_config: Dict[str, Any] = Field(default_factory=dict)
     client_kwargs: Dict[str, Any] = Field(default_factory=dict)
 
-    _client = PrivateAttr()
+    _client: weaviate.WeaviateClient = PrivateAttr()
+    _aclient: weaviate.WeaviateAsyncClient = PrivateAttr()
 
     def __init__(
         self,
         weaviate_client: Optional[Any] = None,
+        weaviate_aclient: Optional[Any] = None,
```

Comment:
Have you considered passing the async client to the same `weaviate_client` parameter and check for the type? Something like
```py
if isinstance(weaviate_client, weaviate.WeaviateClient):
  self._client = weaviate_client
  self._aclient = None
elif isinstance(weaviate_client, weaviate.WeaviateAsyncClient):
  self._client = None
  self._aclient = weaviate_client
```

Reply:
I've changed this to to your suggested implementation now in https://github.com/run-llama/llama_index/pull/17220/commits/cfc9364fef221643d6467b44c52333e1ad4a94be.

This way it is not possible anymore to provide both a sync as well as an async client, but I would not think there are many use cases that require this.